modules/transcription-service-microsoft-azure/src/main/java/org/opencastproject/transcription/microsoftazure/MicrosoftAzureTranscriptionService.java

```diff
@@ -0,0 +1,941 @@
+  public static final String WORKFLOW_CONFIG = "workflow";
+  public static final String DISPATCH_WORKFLOW_INTERVAL_CONFIG = "workflow.dispatch.interval";
+  public static final String COMPLETION_CHECK_BUFFER_CONFIG = "completion.check.buffer";
+  public static final String MAX_PROCESSING_TIME_CONFIG = "max.processing.time";
+  public static final String NOTIFICATION_EMAIL_CONFIG = "notification.email";
+  public static final String CLEANUP_RESULTS_DAYS_CONFIG = "cleanup.results.days";
+  public static final String ENCODING_EXTENSION = "encoding.extension";
+
+  /**
+   * Service configuration values
+   */
+  private boolean enabled = false; // Disabled by default
+  private String subscriptionKey;
+  private String region;
+  private String defaultLanguage = DEFAULT_LANGUAGE;
+  private String workflowDefinitionId = DEFAULT_WF_DEF;
+  private long workflowDispatchInterval = DEFAULT_DISPATCH_INTERVAL;
+  private long completionCheckBuffer = DEFAULT_COMPLETION_BUFFER;
+  private long maxProcessingSeconds = DEFAULT_MAX_PROCESSING_TIME;
+  private String toEmailAddress;
+  private int cleanupResultDays = DEFAULT_CLEANUP_RESULTS_DAYS;
+  private String systemAccount;
+  private ScheduledExecutorService scheduledExecutor = Executors.newScheduledThreadPool(2);
+  protected File tmpStorageDir;
+
+  private AudioStreamContainerFormat compressedAudioFormat = DEFAULT_ENCODING;
+  private ProfanityOption profanityOption = DEFAULT_PROFANITY_OPTION;
+  private List<String> phraseList = new ArrayList<>();
+  private boolean useSubRipTextCaptionFormat = false;
+  private String fileFormat = "vtt";
+
+  public MicrosoftAzureTranscriptionService() {
+    super(JOB_TYPE);
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) {
+    // Has this service been enabled?
+    enabled = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), ENABLED_CONFIG).get();
+    if (!enabled) {
+      logger.info("Service disabled. If you want to enable it, please update the service configuration.");
+      return;
+    }
+    // Mandatory API access properties
+    subscriptionKey = OsgiUtil.getComponentContextProperty(cc, SUBSCRIPTION_KEY);
+    region = OsgiUtil.getComponentContextProperty(cc, REGION);
+    if (subscriptionKey == null) {
+      logger.error("SubscriptionKey not set, disabling service.");
+      return;
+    }
+    if (region == null) {
+      logger.error("Region not set, disabling service.");
+      return;
+    }
+
+    // profanity filter to use
+    Option<String> profanityOpt = OsgiUtil.getOptCfg(cc.getProperties(), PROFANITY_OPTION);
+    if (profanityOpt.isSome()) {
+      try {
+        profanityOption = ProfanityOption.valueOf(profanityOpt.get());
+        logger.info("Profanity filter is set to {}", profanityOption);
+      } catch (IllegalArgumentException e) {
+        logger.error("Profanity filter set to illegal value, disabling service.");
+        return;
+      }
+    } else {
+      logger.info("Default profanity filter will be used");
```

Comment:
This log line should make it clear what level of filtering is in use (ie, none).